android/source/VideoLocker/src/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -0,0 +1,213 @@
+package org.edx.mobile.view;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.BaseFragmentActivity;
+import org.edx.mobile.model.api.RegisterResponse;
+import org.edx.mobile.model.registration.RegistrationFieldType;
+import org.edx.mobile.model.registration.RegistrationForm;
+import org.edx.mobile.model.registration.RegistrationFormField;
+import org.edx.mobile.module.analytics.ISegment;
+import org.edx.mobile.task.RegisterTask;
+import org.edx.mobile.view.custom.ETextView;
+import org.edx.mobile.view.registration.IRegistrationFieldView;
+import org.edx.mobile.view.registration.RegistrationEditTextView;
+import org.edx.mobile.view.registration.RegistrationSpinnerView;
+import org.edx.mobile.view.registration.RegistrationTextAreaView;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class RegisterActivity extends BaseFragmentActivity {
+
+    private RelativeLayout createAccountBtn;
+    private LinearLayout requiredFieldsLayout;
+    private LinearLayout optionalFieldsLayout;
+    private ETextView createAccountTv;
+    private List<IRegistrationFieldView> mFieldViews = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register);
+        overridePendingTransition(R.anim.slide_in_from_bottom, R.anim.no_transition);
+
+        try{
+            segIO.screenViewsTracking(ISegment.Values.LAUNCH_ACTIVITY);
+        }catch(Exception e){
+            logger.error(e);
+        }
+
+        createAccountBtn = (RelativeLayout) findViewById(R.id.createAccount_button_layout);
+        createAccountBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createAccount();
+            }
+        });
+
+        createAccountTv = (ETextView) findViewById(R.id.create_account_tv);
+        requiredFieldsLayout = (LinearLayout) findViewById(R.id.required_fields_layout);
+        optionalFieldsLayout = (LinearLayout) findViewById(R.id.optional_fields_layout);
+        final ETextView optional_text=(ETextView)findViewById(R.id.optional_field_tv);
+        optional_text.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if(optional_text.getText().toString().equalsIgnoreCase(getString(R.string.show_optional_text))) {
```

Comment:
Instead of comparing the text, which seems error prone, can we just check the visibility of the ``optionalFieldsLayout``?